src/v/cloud_storage/segment_chunk_api.h

```diff
@@ -167,4 +167,24 @@ std::unique_ptr<chunk_eviction_strategy> make_eviction_strategy(
   uint64_t max_chunks,
   uint64_t hydrated_chunks);
 
+class segment_chunk_range {
+public:
```

Comment:
I might be missing something, but given the public API of this, could this be a deque of pairs?

Reply:
Do you mean for storing the data inside this class (instead of a map), or for replacing this class entirely?

If the former, then probably yes it can be a deque of pairs and since we only ever iterate over it and never lookup keys themselves, it should be marginally faster, but I don't see that as enough of a speedup to change it (traversal through a tree should still be pretty fast).

If the latter, this class provides some convenience methods for calculating the bounds of the range to decide how much space to reserve in cache etc, it could maybe be done by free functions accepting the deque of pairs but I prefer a class.